src/app/main/main.component.ts

```diff
@@ -290,6 +287,24 @@ export class MainComponent implements OnInit, OnDestroy {
     return dryRun;
   }
 
+  // 'count' is used to track the number of request. Each request is 10MB, so we are limiting the total to 250 MB.
+  getResults(count: number, projectId: string, inputPageToken: string, location: string, jobId: string)  : Promise<BigQueryResponse> {
+    if (!inputPageToken || count >= 25) {
+      // Force an update feature here since everything is done.
+      var localRows : Array<Object> = [];
+      localRows.push(...this.rows);
+      this.rows = localRows;
```

Comment:
This can be a one-liner:

```ts
this.rows = this.rows.slice();
```

But, is it necessary to copy the data to force the update? Would `this._changeDetectorRef.detectChanges();` work?

Reply:
slice() works. I didn't know about it.

We need a copy here because I made changes to the map service to batch detectChanges. In this scenario, we are doing the final change so we want to force an update.